lib/matplotlib/backend_bases.py

```diff
@@ -2574,6 +2574,47 @@ def button_press_handler(event, canvas=None, toolbar=None):
             toolbar.forward()
 
 
+def scroll_handler(event, canvas=None, toolbar=None):
+    ax = event.inaxes
+    if ax is None:
+        return
+    if ax.name != "rectilinear":
+        # zooming is currently only supported on rectilinear axes
+        return
+
+    if toolbar is None:
+        toolbar = (canvas or event.canvas).toolbar
+
+    if toolbar is None or toolbar.mode == _Mode.NONE:
+        return
+
+    if event.key is None:  # zoom towards the mouse position
+        toolbar.push_current()
+
+        xmin, xmax = ax.get_xlim()
+        ymin, ymax = ax.get_ylim()
+        xmin, ymin = ax.transScale.transform((xmin, ymin))
+        xmax, ymax = ax.transScale.transform((xmax, ymax))
+
+        # mouse position in scaled (e.g., log) data coordinates
+        x, y = ax.transScale.transform((event.xdata, event.ydata))
+
+        scale_factor = 1.0 - 0.1 * event.step
```

Comment:
I think the scale factor should be something like `1.1**event.step` (or `0.9**event.step`) so that scrolling one up and one down restores the original view.